### PR TITLE
Added version attribute to components

### PR DESCRIPTION
**Pull request by easylo**
add version attribut for component and test

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Ok, can you explain the benefit of this to me please? Why would you use this over just simply adding the version into the Components name or description fields?

**Comment by easylo** [comment]
on CI process, i need to know the version for each Components. the version value is updated by delivery tools

**Comment by jbrooksuk** [comment]
Ok, I'll give it a proper review tomorrow :)

**Comment by jbrooksuk** [comment]
:beers:

**Comment by easylo** [comment]
thanks

File: app/Models/Component.php [hunk 1]
```diff
@@ -179,4 +182,18 @@ public function getPresenterClass()
     {
         return ComponentPresenter::class;
     }
+
+    /**
+     * Get the presenter class.
+     *
+     * @return string
+     */
+    public function displayVersion()
```
Comment (jbrooksuk) [inline app/Models/Component.php]: This belongs on the `ComponentPresenter`.

File: app/Models/Component.php [hunk 2]
```diff
@@ -179,4 +182,18 @@ public function getPresenterClass()
     {
         return ComponentPresenter::class;
     }
+
+    /**
```
Comment (jbrooksuk) [inline app/Models/Component.php]: The docblock is wrong here.

File: database/migrations/2016_02_16_223700_AlterTableComponentsAddVersionColumn.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('components', function (Blueprint $table) {
+            $table->string('version');
```
Comment (jbrooksuk) [inline database/migrations/2016_02_16_223700_AlterTableComponentsAddVersionColumn.php]: Shouldn't this be `dropColumn`?

File: resources/views/dashboard/components/add.blade.php [hunk 4]
```diff
@@ -33,6 +33,10 @@
                             <label>{{ trans('forms.components.description') }}</label>
                             <textarea name="component[description]" class="form-control" rows="5"></textarea>
                         </div>
+                        <div class="form-group">
+                            <label for="component-version">{{ trans('forms.components.version') }}</label>
+                            <input type="text" class="form-control" name="component[version]" id="component-version" >
```
Comment (jbrooksuk) [inline resources/views/dashboard/components/add.blade.php]: Extra space before the `>`.

File: resources/views/dashboard/index.blade.php [hunk 5]
```diff
@@ -25,7 +25,7 @@
                             <form class='component-inline form-vertical' data-messenger="{{trans('dashboard.components.edit.success')}}">
                                 <div class="row striped-list-item">
                                     <div class="col-lg-4 col-md-3 col-sm-12">
-                                        <h4>{{ $component->name }}</h4>
+                                        <h4>{{ $component->name }} {{ $component->displayVersion() }}</h4>
```
Comment (jbrooksuk) [inline resources/views/dashboard/index.blade.php]: Once the `displayVersion` method is in the presenter, you'll need to replace this with just `$component->display_version`.

Better yet, create a new localization line for it, and you can have it in there, formatted however you want.

File: resources/views/partials/component.blade.php [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <li class="list-group-item {{ $component->group_id ? "sub-component" : "component" }}">
     @if($component->link)
-    <a href="{{ $component->link }}" target="_blank" class="links">{{ $component->name }}</a>
+    <a href="{{ $component->link }}" target="_blank" class="links">{{ $component->name }} {{ $component->displayVersion() }}</a>
```
Comment (jbrooksuk) [inline resources/views/partials/component.blade.php]: Same here (presenter).

File: resources/views/partials/component.blade.php [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
     @else
-    {{ $component->name }}
+    {{ $component->name }} {{ $component->displayVersion() }}
```
Comment (jbrooksuk) [inline resources/views/partials/component.blade.php]: And here.

**Comment by jbrooksuk** [comment]
Can you fix up all tests please?

**Comment by jbrooksuk** [comment]
Closing this due to inactivity.

**Pull request closed without merging**